### PR TITLE
fix(metrics): remove org_id, client_id and user_id tags from operational metrics

### PR DESCRIPTION
**Pull request by Neeraj-swarnkar**
# COMPLETES [#SPARK-403398](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-403398)

## This pull request addresses

removed org_id, client_id and user_id tags from operational metrics,
added client_id as part of fields instead of tags
added/updated unit test for the same

## by making the following changes

files - 
packages/@webex/internal-plugin-metrics/src/metrics.js
packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js
removed org_id, client_id and user_id tags from operational metrics,
added client_id as part of fields instead of tags
added/updated unit test for the same

`client_id` has been added to the field, PFA for the same as proof value has been passed in payload.

image

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-metrics/src/metrics.js [hunk 1]
```diff
@@ -69,23 +69,16 @@ const Metrics = WebexPlugin.extend({
       // eslint-disable-next-line no-undef
       domain:
         typeof window !== 'undefined' ? window.location.hostname || 'non-browser' : 'non-browser', // Check what else we could measure
-      client_id: this.webex.credentials.config.client_id,
```
Comment (Neeraj-swarnkar) [inline packages/@webex/internal-plugin-metrics/src/metrics.js:72]: removed org_id, client_id and user_id tags from operational metrics,

File: packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js [hunk 2]
```diff
@@ -199,17 +199,15 @@ describe('plugin-metrics', () => {
...
               assert.property(metric.tags, 'domain');
-              assert.property(metric.tags, 'client_id');
-              assert.property(metric.tags, 'user_id');
 
               assert.property(metric.fields, 'browser_version');
               assert.property(metric.fields, 'os_version');
               assert.property(metric.fields, 'sdk_version');
               assert.property(metric.fields, 'platform');
               assert.property(metric.fields, 'spark_user_agent');
+              assert.property(metric.fields, 'client_id');
```
Comment (Neeraj-swarnkar) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js:210]: Updated unit test for the changes

File: packages/@webex/internal-plugin-metrics/src/metrics.js [hunk 3]
```diff
@@ -69,23 +69,16 @@ const Metrics = WebexPlugin.extend({
...
-      user_id: this.webex.internal.device.userId,
```
Comment (meta-dipanshu-sharma) [inline packages/@webex/internal-plugin-metrics/src/metrics.js:73]: as we're removing `device.userId` so means not using `@webex/internal-plugin-device` package. So please remove this import from index.js file as well as this is causing a circular dependency with `@webex/internal-plugin-metrics` package.

File: packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js (hunk 2 above)
Comment (meta-dipanshu-sharma) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js:210]: please remove `device: {userId: 'userId'}` from line 92 object as now we're not using it.

**Review by meta-dipanshu-sharma — CHANGES_REQUESTED** [review]

File: packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js (hunk 2 above)
Comment (Neeraj-swarnkar) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/metrics.js:210]: Removed, @meta-dipanshu-sharma Please check.

File: packages/@webex/internal-plugin-metrics/src/metrics.js (hunk 3 above)
Comment (Neeraj-swarnkar) [reply packages/@webex/internal-plugin-metrics/src/metrics.js:73]: Thanks for pointing out. I have removed, @meta-dipanshu-sharma Please check.

**Review by meta-dipanshu-sharma — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -78,17 +78,13 @@ webex.rooms.create({ title: `My First Room` }).then(room => {
...
+We provide a built, minified version of the SDK, that includes `window.Webex`. You can access it via [unpkg](https://unpkg.com/) or [jsdelivr](https://jsdelivr.com/).
 
 ```html
 <!-- unpkg -->
 <script crossorigin src="https://unpkg.com/webex/umd/webex.min.js"></script>
 <!-- jsdelivr -->
 <script crossorigin src="https://cdn.jsdelivr.net/npm/webex/umd/webex.min.js"></script>
-<!-- gitcdn -->
-<script
-  crossorigin
```
Comment (Neeraj-swarnkar) [inline README.md:90]: Fixed broken link as part of this PR only.

File: packages/webex/README.md [hunk 5]
```diff
@@ -65,15 +65,13 @@ webex.rooms.create({ title: `My First Room` }).then(room => {
...
-We provide a built, minified version of the SDK, that includes `window.Webex`. You can access it via [unpkg](https://unpkg.com/), [jsdelivr](https://jsdelivr.com/), or [gitcdn.xyz](https://gitcdn.xyz/).
+We provide a built, minified version of the SDK, that includes `window.Webex`. You can access it via [unpkg](https://unpkg.com/) or [jsdelivr](https://jsdelivr.com/).
 
 ```html
 <!-- unpkg -->
 <script src="https://unpkg.com/webex/umd/webex.min.js"></script>
 <!-- jsdelivr -->
 <script src="https://cdn.jsdelivr.net/npm/webex/umd/webex.min.js"></script>
-<!-- gitcdn -->
-<script src="https://gitcdn.xyz/repo/webex/webex-js-sdk/master/packages/node_modules/webex/umd/webex.min.js"></script>
```
Comment (Neeraj-swarnkar) [inline packages/webex/README.md:76]: Removed broken link from the readme file.

**Review by arun3528 — APPROVED** [review]

**Review by marcin-bazyl — APPROVED** [review]

**Comment by marcin-bazyl** [comment]
what about beta branch?

**Comment by Neeraj-swarnkar** [comment]
> what about beta branch?

I will cherry pick the changes and raise the PR in Beta.

**Pull request merged**